test/Test/Pos/Ssc/GodTossing/ComputeSharesSpec.hs

```diff
@@ -9,31 +9,60 @@ import           Test.Hspec.QuickCheck   (prop)
 import           Universum
 
 import           Control.Monad.Except    (runExcept)
-import qualified Data.HashMap.Strict     as HM (elems, keys)
-import           Pos.Types.Coin          (mkCoin)
-import           Pos.Lrc                 (RichmenStake)
+import qualified Data.HashMap.Strict     as HM
+import           Pos.Constants           (genesisMpcThd)
+import           Pos.Types.Coin          (mkCoin, sumCoins)
+import           Pos.Types.Core          (coinPortionDenominator, getCoinPortion)
+import           Pos.Lrc                 (InvalidRichmenStake (..), RichmenStake,
+                                          ValidRichmenStake (..))
 import qualified Pos.Ssc.GodTossing      as T
-import           Test.QuickCheck         (Property, (==>))
 
 spec :: Spec
 spec = describe "computeSharesDistr" $ do
     prop emptyRichmenStakeDesc emptyRichmenStake
-    -- prop allRichmenGetShareDesc allRichmenGetShares
+    prop allRichmenGetShareDesc allRichmenGetShares
+    prop invalidStakeErrorsDesc invalidStakeErrors
+    prop totalStakeZeroDesc totalStakeIsZero
+    prop validRichmenStakeWorksDesc validRichmenStakeWorks
   where
     emptyRichmenStakeDesc = "Fails to calculate a share distribution when the richmen\
     \ stake is empty."
-    allRichmenGetShareDesc = "All richmen are awarded a share, and richmen who do not\
-    \ participate in the share distribution are awarded none"
+    allRichmenGetShareDesc = "All richmen are awarded a non-zero share, and richmen who\
+    \ do not participate in the share distribution are awarded none"
+    invalidStakeErrorsDesc = "If the stake distribution has a 'StakeholderId' with\
+    \ insufficient stake to participate in the distribution of shares, the distribution\
+    \ fails to be calculated."
+    totalStakeZeroDesc = "If the total stake is zero, then the distribution fails to be\
+    \ calculated"
+    validRichmenStakeWorksDesc = "Given a valid distribution of stake, calculating the\
+    \ distribution of shares successfully works."
 
 emptyRichmenStake :: RichmenStake -> Bool
 emptyRichmenStake richmen =
     let zeroCoin = mkCoin 0
         allZero = runExcept . T.computeSharesDistr . fmap (const zeroCoin) $ richmen
     in isLeft allZero
 
-allRichmenGetShares :: RichmenStake -> Property
-allRichmenGetShares richmen =
+allRichmenGetShares :: ValidRichmenStake -> Bool
+allRichmenGetShares (getValid -> richmen) =
     let outputStakeholder = runExcept $ T.computeSharesDistr richmen
-    in isRight outputStakeholder ==>
-        ((Right . HM.keys $ richmen) == (HM.keys <$> outputStakeholder)) &&
-        (either (const False) (all (/= 0) . HM.elems) outputStakeholder)
+    in ((Right . HM.keys $ richmen) == (HM.keys <$> outputStakeholder)) &&
+        (either (const False) (all (/= 0)) outputStakeholder)
```

Comment:
I think it would be more readable if you wrote it using `case`. Something like
```
case runExcept $ T.compute… of 
  Left _ -> False
  Right outputStakeholder -> …
```
In this case you wouldn't need to use `Right` and `<$>` and `either (const False)`.